Amplify/Core/Support/AmplifyTask+OperationTaskAdapters.swift

```diff
@@ -20,7 +20,9 @@ public class AmplifyOperationTaskAdapter<Request: AmplifyOperationRequest,
     public init(operation: AmplifyOperation<Request, Success, Failure>) {
         self.operation = operation
         self.childTask = ChildTask(parent: operation)
-        resultToken = operation.subscribe(resultListener: resultListener)
+        resultToken = operation.subscribe { [weak self] in
+            self?.resultListener($0)
+        }
```

Comment:
this seems important enough to ship on its own outside of this PR

Reply:
yeah, this seems like a mem leak. Will create a separate PR for it.